components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -1096,6 +1096,49 @@ public String getServiceProviderNameByClientId(String clientId, String clientTyp
 
     }
 
+    /**
+     * @param inboundKey
+     * @param inboundType
+     * @param tenantDomain
+     * @return
+     * @throws IdentityApplicationManagementException
+     */
+    @Override
+    public String getApplicationResourceIDByInboundKey(String inboundKey, String inboundType,
+                                                       String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        // invoking the listeners
+        Collection<ApplicationMgtListener> listeners = getApplicationMgtListeners();
+        for (ApplicationMgtListener listener : listeners) {
+            if (listener.isEnable() && !listener.doPreGetApplicationResourceIDByInboundKey(inboundKey, inboundType,
+                    tenantDomain)) {
+                return null;
+            }
+        }
+
+        ApplicationDAO appDAO = ApplicationMgtSystemConfig.getInstance().getApplicationDAO();
```

Comment:
should we start and end a tenant flow as done in other methods?

Reply:
Check in the underlying methods in this block rely on tenantID set in the carbonContext. If not we can avoid it.